src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrUpdater.cs

```diff
@@ -320,8 +325,8 @@ public class VmrUpdater : VmrManagerBase, IVmrUpdater
 
             foreach (var (dependency, dependencyMapping) in await GetDependencies(mappingToUpdate, cancellationToken))
             {
-                if (updatedDependencies.Any(d => d.Mapping == dependencyMapping) ||
-                    skippedDependencies.Any(d => d.Mapping == dependencyMapping))
+                var processedDependencies = reposToUpdate.Concat(updatedDependencies).Concat(skippedDependencies);
```

Comment:
We don't have to do this every loop, we could initialize it outside

Reply:
And I guess we'd need to update it alongside `reposToUpdate`, but still should be a bit faster

Reply:
So, this is a view only (`IEnumerable`) which would stop working each time one of these collections change because the enumerator would become invalid. I modify two of those (`skippedDependencies` and `reposToUpdate`).

Since there's only <30 items in total, I think it's much more readable and less error prone to create a new enumerable always (I am not even allocating anything outside of the enumerator).

Reply:
Yeah, I agree this is the most readable solution and the collections are always small